js/net.ts

```diff
@@ -81,18 +90,28 @@ class ConnImpl implements Conn {
    */
   closeRead(): void {
     // TODO(ry) Connect to AsyncWrite::shutdown in resources.rs
-    return notImplemented();
+    shutdown(this.fd, false);
   }
 
   /** closeWrite shuts down (shutdown(2)) the writing side of the TCP
    * connection. Most callers should just use close().
    */
   closeWrite(): void {
     // TODO(ry) Connect to AsyncWrite::shutdown in resources.rs
-    return notImplemented();
+    shutdown(this.fd, true);
   }
 }
 
+function shutdown(fd: number, isWrite: boolean) {
+  const builder = new flatbuffers.Builder();
+  msg.Shutdown.startShutdown(builder);
+  msg.Shutdown.addRid(builder, fd);
+  msg.Shutdown.addIsWrite(builder, isWrite);
+  const inner = msg.Shutdown.endShutdown(builder);
+  const baseRes = dispatch.sendSync(builder, msg.Any.Shutdown, inner);
```

Comment:
Not sure this should be sync. @piscisaureus do you know?

Reply:
Let's assume it doesn't until proven otherwise. It would be nice if this was not async.